jsapp/js/actions/datashare.es6

```diff
@@ -0,0 +1,110 @@
+/**
+ * Dynamic data attachment related actions
+ */
+
+import Reflux from 'reflux';
+import alertify from 'alertifyjs';
+import {dataInterface} from 'js/dataInterface';
+import {MAX_DISPLAYED_STRING_LENGTH} from '../constants';
+import {
+  getAssetUIDFromUrl,
+  truncateFile,
+  truncateString,
+} from '../utils';
+
+const dataShareActions = Reflux.createActions({
+  attachToSource: {children: ['completed', 'failed']},
+  detachSource: {children: ['completed', 'failed']},
+  patchSource: {children: ['completed', 'failed']},
+  getAttachedSources: {children: ['completed', 'failed']},
+  getSharingEnabledAssets: {children: ['completed', 'failed']},
+  toggleDataSharing: {children: ['completed', 'failed']},
+  updateColumnFilters: {children: ['completed', 'failed']},
+});
+
+dataShareActions.attachToSource.listen((assetUid, data) => {
+  dataInterface.attachToSource(assetUid, data)
+    .done(dataShareActions.attachToSource.completed)
+    .fail(dataShareActions.attachToSource.failed);
+});
+dataShareActions.attachToSource.failed.listen((response) => {
+  alertify.error(response?.responseJSON || t('Failed to attach to source'));
+});
+
+dataShareActions.detachSource.listen((attachmentUrl) => {
+  dataInterface.detachSource(attachmentUrl)
+    .done(dataShareActions.detachSource.completed)
+    .fail(dataShareActions.detachSource.failed);
+});
+dataShareActions.detachSource.failed.listen((response) => {
+  alertify.error(response?.responseJSON || t('Failed to detach from source'));
+});
+
+dataShareActions.patchSource.listen((attachmentUrl, data) => {
+  dataInterface.patchSource(attachmentUrl, data)
+    .done(dataShareActions.patchSource.completed)
+    .fail(dataShareActions.patchSource.failed)
+});
+dataShareActions.patchSource.failed.listen((response) => {
+  alertify(response?.responseJSON || t('Failed to patch source'));
+});
+
+dataShareActions.getAttachedSources.listen((assetUid) => {
+  dataInterface.getAttachedSources(assetUid)
+    .done((response) => {
+      let allSources = [];
+      response.results.forEach((source) => {
```

Comment:
I'm not sure, I've left a comment to investigate this later but this would only pop up if there are more than 100 connected projects which I feel like may be an edge case